src/cogent3/core/new_alignment.py

```diff
@@ -2614,28 +2701,53 @@ def __getitem__(self, index: Union[str, int]) -> Aligned:
 
     @__getitem__.register
     def _(self, index: str) -> Aligned:
-        adv = self.get_aligned_view(seqid=index)
-        return Aligned(data=adv, moltype=self.moltype)
+        adv = self.get_view(seqid=index)
+        return Aligned(data=adv)
 
     @__getitem__.register
     def _(self, index: int) -> Aligned:
         return self[self.names[index]]
 
     @property
     def names(self) -> tuple[str]:
-        return self._names
+        return list(self._seqs.keys())
 
     @property
     def alphabet(self) -> new_alphabet.CharAlphabet:
         return self._alphabet
 
-    def get_aligned_view(self, seqid: str) -> AlignedDataView:
-        # refactor: design
-        # need to revisit what the variable is called i.e. parent_length
-        return AlignedDataView(seq=self, seqid=seqid, seq_len=self.align_len)
+    @property
+    def make_seq(self) -> Optional[MakeSeqCallable]:
+        """if set, returns a function that takes 'seq' and 'name' as keyword
+        arguments and returns a corresponding Sequence from the collection.
+
+        Notes
+        -----
+        Can be set with any callable function that takes 'seq' and 'name' as
+        keyword arguments. Typically set with '<moltype-instance>.make_seq'.
+        """
+        return self._make_seq
+
+    @make_seq.setter
+    def make_seq(self, make_seq: MakeSeqCallable) -> None:
+        self._make_seq = make_seq
+
+    @property
+    def align_len(self) -> int:
+        return self._align_len
+
+    def seq_lengths(self) -> dict[str, int]:
```

Comment:
it should return the length of the sequences, not the alignment length. Look at the method on `AlignmentI.get_seq_lengths()`